pyccel/stdlib/ndarrays/ndarrays.c

```diff
@@ -371,5 +424,45 @@ int64_t     *numpy_to_ndarray_shape(int64_t *np_shape, int nd)
     for (int i = 0; i < nd; i++)
         nd_shape[i] = np_shape[i];
     return nd_shape;
+}
 
+int get_shape_product(int64_t *shape, int nd, int max_nd)
+{
+    int product = 1;
+
+    for (int i = nd; i < max_nd; ++i)
+        product *= shape[i];
+    return (product);
+}
+
+int element_index(t_ndarray arr, uint32_t element_num, int nd)
+{
+    if (arr.order == order_c)
+        return element_num;
+    if (nd == 0)
+        return (0);
+    if (nd == arr.nd)
+        return (element_num % arr.shape[nd - 1]) * arr.strides[nd - 1] + element_index(arr, element_num, nd - 1);
+    int true_index = (element_num / (get_shape_product(arr.shape, nd, arr.nd)));
+    if (true_index >= arr.shape[nd - 1])
+        true_index = true_index % arr.shape[nd - 1];
+    return (true_index * arr.strides[nd - 1] + element_index(arr, element_num, nd - 1));
+}
+
+void array_copy_data(t_ndarray *dest, t_ndarray src)
+{
+    unsigned char *d = (unsigned char*)dest->raw_data;
+    unsigned char *s = (unsigned char*)src.raw_data;
+
+    if (dest->order == src.order && dest->order == order_c)

```

Comment:
I think checking for `order_c` for both arrays can be easier for later if we want to change some code.
```suggestion
    if (dest->order == order_c && src.order == order_c)
```

Reply:
I don't see why this change would make things easier if we want to make changes? Can you explain your reasoning?

Reply:
i just think comparing both `src.order` and `dest->order` to `order_c` separately is more easier to understand if in the future we try to add something to that condition.